src/ghostwriter/handlers/external.py

```diff
@@ -50,3 +55,24 @@ async def get_index(
         application_name=config.name,
     )
     return Index(metadata=metadata)
+
+
+@external_router.api_route(
+    "/rewrite/{full_path:path}", response_class=RedirectResponse
+)
+async def rewrite(
+    full_path: Annotated[str, Path(title="The URL path to rewrite")],
+    request: Request,
+    logger: Annotated[BoundLogger, Depends(logger_dependency)],
+    context: Annotated[RequestContext, Depends(context_dependency)],
+) -> str:
+    logger.debug(f"Request for rewrite: {full_path} [{request.method}]")
+    params = Parameters(
+        user=context.user,
+        token=context.token,
+        client=context.rsp_client,
+        base_url=str(context.factory.context.base_url),
```

Comment:
This seems like a bit of deep nesting to get the `base_url`, is it perhaps easier if you define a
config_dependency and you can then import and use as:

    base_url = str(config.base_url)

Or make config property of the Request context? Then you can use as:

    context.config.base_url

Reply:
I already have a config dependency, I just didn't use it.  Might as well.